### PR TITLE
[DNM] Zephyr smp rework

### PR DESCRIPTION
**Pull request by RanderWang**
Update secondary core setting based on zephyr smp changes https://github.com/zephyrproject-rtos/zephyr/pull/64755.

Waiting for : https://github.com/thesofproject/sof/pull/8747

### CODE REVIEW

File: zephyr/lib/cpu.c [hunk 1]
```diff
@@ -149,19 +123,13 @@ int cpu_enable_core(int id)
 	 * and the idle thread stack.
 	 */
 	if (pm_state_next_get(id)->state == PM_STATE_ACTIVE)
-		z_init_cpu(id);
+		cpu_resume = false;
```
Comment (lyakh) [inline zephyr/lib/cpu.c:126]: could just do
```
	if (pm_state_next_get(id)->state == PM_STATE_ACTIVE) {
		k_smp_cpu_start(id, secondary_init, NULL);
		return 0;
	}
```
here and avoid adding `cpu_resume` altogether.

Comment (RanderWang) [reply zephyr/lib/cpu.c:126]: need to consider #if ZEPHYR_VERSION(3, 0, 99) <= ZEPHYR_VERSION_CODE

Comment (RanderWang) [reply zephyr/lib/cpu.c]: thanks, updated

File: zephyr/lib/cpu.c [hunk 2]
```diff
@@ -239,29 +206,24 @@ int cpu_enable_core(int id)
...
-
-	atomic_clear(&start_flag);
-	atomic_clear(&ready_flag);
-
-	arch_start_cpu(id, z_interrupt_stacks[id], CONFIG_ISR_STACK_SIZE,
-		       secondary_init, &start_flag);
-
-	while (!atomic_get(&ready_flag))
-		k_busy_wait(100);
+	/* During kernel initialization, the next pm state is set to ACTIVE. By checking this
```
Comment (kv2019i) [inline zephyr/lib/cpu.c:213]: Should we remove the ZEPHYR_VERSION check? Afer this change, we won't support older Zephyr 3.0 anymore in SOF main, so might as well drop this ifdef.

File: west.yml [hunk 3]
```diff
@@ -43,7 +43,7 @@ manifest:
 
     - name: zephyr
       repo-path: zephyr
-      revision: d7af6f371034f31b9440b27c694b0be3c87491d3
+      revision: 04de43b1a119665532b6b2a0c6bbcbefd47eb7d6
```
Comment (kv2019i) [inline west.yml:46]: For the mergeable version, this needs to be combined with the previous git commit. Otherwise git-bisect will be broken as the first commit cannot be compiled with old zephyr so this has to be a single git commit to change the API and modify west.yml.

**Review by kv2019i — CHANGES_REQUESTED** [review]
@RanderWang can you check why TGL builds are failing? https://sof-ci.01.org/sofpr/PR8764/build2107/build/index.html

This shouldn't happen and now we don't get any test results for this version. If this would pass the CI, we could just merge this and drop the older Zephyr baseline update PR.

File: west.yml (hunk 3 above)
Comment (RanderWang) [reply west.yml]: thanks

File: zephyr/lib/cpu.c (hunk 2 above)
Comment (RanderWang) [reply zephyr/lib/cpu.c:208]: sure, thanks

**Comment by tmleman** [comment]
> QuickBuild test case pass, but got failed result, why ?

@RanderWang I've reviewed the last four builds for this pull request on QB. In the older ones, TGL doesn't build, but in the latest one, this issue is no longer present. However, there is a failure in the gain_quality test (also TGL). The log indicates that there was an issue in the final step during the audio quality verification. @wszypelt , could you check if this is a problem with the test?

PS. I've scheduled a rebuild and rerun of tests.

**Review by kv2019i — APPROVED** [review]
Mandatory tests are now passing. The MTL error is known, so only open now is the cavs2.5 fails and whether there is more of them with this update than we have in baseline (we do have fails with pause-resume in mainline as well).

UPDATE: e.g. in this recent PR test, all these cavs2.5 tests pass:
https://sof-ci.01.org/sofpr/PR8756/build1998/devicetest/index.html
Compared to this PR:
https://sof-ci.01.org/sofpr/PR8764/build2195/devicetest/index.html

File: west.yml [hunk 4]
```diff
@@ -43,7 +43,7 @@ manifest:
 
     - name: zephyr
       repo-path: zephyr
-      revision: d7af6f371034f31b9440b27c694b0be3c87491d3
+      revision: 6a0b1da158a4f8bc5f2ca5058637dce26a38660e
```
Comment (lgirdwood) [inline west.yml:46]: can you make this a separate patch for bisect-ability.

**Review by lgirdwood — COMMENTED** [review]
LGTM, just need to split the west update into another patch.

File: west.yml (hunk 4 above)
Comment (tmleman) [reply west.yml:46]: I think that if we separate the west update into a separate commit, then we'll lose the ability to bisect because these changes won't compile without updating Zephyr.
@RanderWang  correct me if I'm wrong.

Comment (kv2019i) [reply west.yml:46]: @lgirdwood west update needs to be in same commit, otherwise bisect is broken.

**Comment by kv2019i** [comment]
Marking as DNM so this doesn't get accidentally merged. We need to understand first why we have so many fails in https://sof-ci.01.org/sofpr/PR8764/build2195/devicetest/index.html

**Comment by kv2019i** [comment]
https://github.com/thesofproject/sof/issues/8792 filed for the chain-dma failures seen. I'll submit a fix soon to Zephyr.

**Comment by kv2019i** [comment]
https://github.com/thesofproject/sof/pull/8790 submitted to fix the double-free.
